src/libraries/System.Net.Http.Json/ref/System.Net.Http.Json.cs

```diff
@@ -0,0 +1,43 @@
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PostAsJsonAsync<T>(this System.Net.Http.HttpClient client, System.Uri requestUri, T value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PutAsJsonAsync(this System.Net.Http.HttpClient client, string requestUri, System.Type type, object? value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PutAsJsonAsync(this System.Net.Http.HttpClient client, System.Uri requestUri, System.Type type, object? value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PutAsJsonAsync<T>(this System.Net.Http.HttpClient client, string requestUri, T value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PutAsJsonAsync<T>(this System.Net.Http.HttpClient client, System.Uri requestUri, T value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+    }
+    public static partial class HttpContentJsonExtensions
+    {
+        public static System.Threading.Tasks.Task<object> ReadFromJsonAsync(this System.Net.Http.HttpContent content, System.Type type, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<T> ReadFromJsonAsync<T>(this System.Net.Http.HttpContent content, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+    }
+    public partial class JsonContent : System.Net.Http.HttpContent
+    {
+        public JsonContent(System.Type type, object? value, System.Net.Http.Headers.MediaTypeHeaderValue mediaType, System.Text.Json.JsonSerializerOptions options = null) { }
+        public JsonContent(System.Type type, object? value, string mediaType, System.Text.Json.JsonSerializerOptions options = null) { }
+        public JsonContent(System.Type type, object? value, System.Text.Json.JsonSerializerOptions options = null) { }
```

Comment:
I understand we've had a security concern about just using GetType, for the core methods on JsonSerializer.  Do we need to have the same concern here, or could we have an overload that just takes `object value` rather than `Type type, object? value`?

Reply:
I'd say especially here because you clearly send the data to a different machine. /cc @GrabYourPitchforks @bartonjs

Reply:
But also presumably with a very clear schema. I understand the theoretical nature of it, but I think we may be taking the security concerns too far here. Yes, you could send a UserWithSalary when only a User was needed. Are we really thwarting that here? I guarantee you devs are just going to end up doing `obj.GetType(), obj`, and we will have achieved no additional security and more complicated code, woohoo.